### ISSUE TITLE
Best practice about pages/views structuration

### ISSUE THREAD

**Issue by mlenormand** [rstoenescu/quasar-framework#30]
Not a bug, but rather a discussion about pages/views structuration (OK to use Github issues for that or you prefer a more appropriate place ?)

Putting all the components in the /components folder is not convenient because the number of components (layouts, views, ui blocks, custom widgets, ...) quickly grows. What I did in my app is using different folders : /components, /layouts, /pages. Initial version of Quasar was also based on multiple folders (and one folder for each artefact, which was quite "heavy", the use of .vue files in 0.5.0 was a great improvement !).

Extract from my router.js file : 

```
let routes = {
  // Default
  '/': {
    component: require('layouts/LayoutMain.vue'),
    subRoutes: {
      '/search': {component: require('pages/PageSearch.vue')},
      '/item': {component: require('pages/PageItem.vue')}
    }
  },
  // Default
  '/login': {
    component: require('layouts/LayoutLogin.vue'),
    subRoutes: {
      '/': {component: require('pages/PageLogin.vue')},
    }
  }
}
```

Extract from my weback.base.conf.js file : 

```
  resolve: {
    extensions: ['', '.js', '.vue'],
    fallback: [path.join(__dirname, '../node_modules')],
    alias: {
      'quasar': path.resolve(__dirname, '../node_modules/quasar-framework/'),
      'src': path.resolve(__dirname, '../src'),
      'assets': path.resolve(__dirname, '../src/assets'),
      'components': path.resolve(__dirname, '../src/components'),
      'layouts': path.resolve(__dirname, '../src/layouts'),
      'pages': path.resolve(__dirname, '../src/pages')
    }
  },
```

**Remark** : customize the existing load() function (in router.js) to pass it for example 'layouts/LayoutMain.vue' or 'pages/PageSearch.vue' doesn't work, certainly because of the statically analysys of the dependencies by Webpack.

About the responsability of layouts, views, what do you think about putting header/footer UI parts in the views rather than in the layouts ? The header includes the title, icon, sometimes subtitles, and those informations are specific to the current view. If we include its to the layout, caching pages would be difficult because each time the user returns to the previous page, we would have to "update" the header/footer part (I currently have this real case on my app !).

**My proposal** : header/footer in the "view", and all the other parts (mainly drawers) in the "layout".
A "view" just have to show the drawer buttons if the layout it belongs has drawer(s).

**Comment by rstoenescu** [comment]
One of key aspects that I wanted in v0.5 is to allow developers to structure their code as they feel like. The original design was too "heavy" with pages, each having its own folder with multiple files.

As the new architecture allows for greater flexibility there comes greater responsibility. The recommended way to go would just as done in Quasar Play app, which is to structure folders based on routes.

The template and `load()` function are there just as a starting point. You as a developer are free to change it however you like.. even delete it and just require each route component.

Since you can structure code to your liking, it's a good thing to use aliases just like you did in the `webpack.base.conf.js` file.

Layouts should contain markup that wrap your pages. I know, title is linked to the page, but there are ways to communicate between components, like Vuex or simply using a "store" just like in Quasar Play app. Take a look.
The reason why we shouldn't include layouts in page views is that:
1. Duplication occurs (which defeats the purpose of layouts). Imagine for example that you would have to duplicate the navigational tabs on each page. Changing the label of one tab would make you go through all pages for this. Maybe you forget one page. Hell comes loose :) Same for that button that you may place on a toolbar. Same for colors and everything else.
2. There's no way to place the Drawer on the left or right side of page view on bigger width windows if toolbars are placed within page templates. The layout uses flexbox to place its components. Take a look in Developer tools.

So to summarize, best way is just like in Quasar app, where a "store" `.js` file was written which controls the toolbars. Include that store in each page and change Toolbars from data within that store.

**Comment by rstoenescu** [comment]
Please reopen if I missed something.
